### PR TITLE
feat: Templates, Template Packs and Plugins first iteration

### PR DESCRIPTION
**Pull request by devcatalin**
This PR...

## Changes

- introduced new redux slice `extension` which keeps the plugins, templates and template packs state
- used the term "Extension" for a general way to refer to either a plugin, a template or a template pack
- added new types: `AnyExtension<ExtensionType>` , `AnyPlugin` , `AnyTemplate` and `TemplatePack` + more specific types for each type of template and plugin
- created `templateService` which handles downloading, loading and updating templates and template packs
- created `pluginService` which handles downloading, loading and updating plugins
- created `TemplateExplorerPane` component to display the list of installed templates and `TemplateInstallModal` for template and template pack installation (download + load)
- created `TemplateModal` and `TemplateFormRenderer` for opening a modal that renders multi-step forms defined in the template
- created `PluginManagerPane` component to display the list of installed plugins and `PluginInstallModal` for plugin installation
- created generic `downloadExtension` and `downloadExtensionEntry` to easily download, validate and save any type of extensions
- added two new tabs to the left menu
- and more...

### Must have
- [x] logic for updating template/plugin versions - @devcatalin 
- [x] check for new template/plugin versions and update them (on application load and manually via a button) - @devcatalin 
- [x] actions to delete template / template pack / plugin - @topliceanurazvan 
- [x] refactor text interpolation for template files (could use regex + lodash or a library like mustache) - @andreiv1992 
- [x] improve UI/UX (could use section blueprints for rendering if we want to group templates by template packs and plugins - @topliceanurazvan 
- [x] make template modal resizable - @topliceanurazvan 

### Nice have
- [ ] strip descriptions if they are too long, add ellipsis
- [ ] first step when opening a template should be a page showing all template information
- [ ] move extensions ipc calls from `main.ts` to separate file 
- [ ] support for vanilla templates - @devcatalin 
- [ ] support for bundled helm chart templates - @devcatalin 
- [ ] opt-in setting for automatically updating all extensions (if new versions are available) when Monokle loads
- [ ] individual actions to update a single template, template pack or plugin
- [ ] improve template modal responsiveness - @topliceanurazvan 
- [ ] categorize templates by sections (standalone, from template pack, from plugin)..
- [ ] add search bar for templates and plugins panes - @topliceanurazvan 
- [ ] JSON kind handlers plugins
- [ ] use the same Form widgets as the resource's form editor
- [ ] allow usage of widgets via json schemas
- [ ] allow multiple github urls and support multiple github branches for downloading extensions
- [ ] pin favorite templates
- [ ] support for template and plugin icons via the `icon` field which points to an image file
- [ ] post-processing to resize and reduce image sizes for icons
- [ ] md5 hash of downloaded extension content to disallow external changes to the files
- [ ] terms of service for using extensions

## How to test it

- Links for installing test extensions:
  - Plugin: https://github.com/devcatalin/monokle-test-plugin
  - Template: https://github.com/devcatalin/monokle-test-template
  - Template Pack: https://github.com/devcatalin/monokle-test-template-pack

## Screenshots

-

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/PluginManagerPane/PluginInformation.tsx
Comment (topliceanurazvan) [inline src/components/organisms/PluginManagerPane/PluginInformation.tsx:65]: Maybe we should put all styled components in a separate `styled` file

File: src/components/organisms/PluginManagerPane/PluginManagerPane.tsx
Comment (topliceanurazvan) [inline src/components/organisms/PluginManagerPane/PluginManagerPane.tsx:50]: Do we need duplicate `Tooltip` here?

File: src/components/organisms/TemplateExplorerPane/TemplateExplorerPane.tsx
Comment (topliceanurazvan) [inline src/components/organisms/TemplateExplorerPane/TemplateExplorerPane.tsx:60]: Same, do we need it duplicated?

File: src/components/organisms/TemplateExplorerPane/TemplateInformation.tsx
Comment (topliceanurazvan) [inline src/components/organisms/TemplateExplorerPane/TemplateInformation.tsx:69]: Same as above, have all styled components in a separate `styled` file?

**Review by topliceanurazvan — COMMENTED** [review]
Left some comments. LGTM!

File: src/components/organisms/PluginManagerPane/PluginInformation.tsx
Comment (devcatalin) [reply src/components/organisms/PluginManagerPane/PluginInformation.tsx:1]: Yes, thanks

File: src/components/organisms/TemplateExplorerPane/TemplateInformation.tsx
Comment (devcatalin) [reply src/components/organisms/TemplateExplorerPane/TemplateInformation.tsx:69]: Yes, will do

File: src/components/organisms/PluginManagerPane/PluginManagerPane.tsx
Comment (devcatalin) [reply src/components/organisms/PluginManagerPane/PluginManagerPane.tsx:50]: Good catch

File: electron/extensions/fileSystem.ts
Comment (devcatalin) [reply electron/extensions/fileSystem.ts:1]: I wanted to use only promises so I can leverage the performance boost from the `async` library

File: electron/extensions/downloadExtension.ts
Comment (devcatalin) [reply electron/extensions/downloadExtension.ts:1]: I'm using `tar` to extract the files from the downloaded github repos. Is there a native solution for this without installing the library?

**Review by olensmar — APPROVED** [review]
LGTM - let's review in Monokle once merged and decide on what needs fixing for 1.5.0

**Pull request merged**
